### PR TITLE
feat(puppeteer): Upgrade to axe-core 4.3

### PR DESCRIPTION
**Pull request by WilcoFiers**

### CODE REVIEW

File: packages/puppeteer/src/axePartialRunner.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import * as axe from 'axe-core';
+
+export type PartialResults = Parameters<typeof axe.finishRun>[0];
+
+/**
+ * This class parallelizes the async calls to axe.runPartial.
+ *
+ * In this project, most async calls needs to block execution, such as
+ * the driver.switchTo().frame() calls. These must run in the correct order
+ * to avoid stale element references and infinite loops testing the same frame.
```
Comment (WilcoFiers) [inline packages/puppeteer/src/axePartialRunner.ts:10]: Oops, copy from WDJS. Will fix.

File: packages/puppeteer/test/axePuppeteer.test.ts [hunk 2]
```diff
@@ -0,0 +1,711 @@
...
+          run: () => new Promise(resolve => resolve({})),
+          configure: () => {}
+        }
+      `;
+      await page.goto(`${addr}/external/index.html`);
+      const evalSpy: SinonSpy = sinon.spy(page.mainFrame(), 'evaluate');
+      await new AxePuppeteer(page, axeSource).analyze();
+      assert(evalSpy.calledWith(axeSource));
+    });
+    // TODO: Defaults to using the bundled axe-core source
```
Comment (straker) [inline packages/puppeteer/test/axePuppeteer.test.ts:108]: Do we need to worry about this TODO in this pr?

File: packages/puppeteer/test/axePuppeteer.test.ts [hunk 3]
```diff
@@ -0,0 +1,711 @@
...
+            tags: ['wcag2aa']
+          }
+        ]
+      };
+
+      await page.goto(`${addr}/external/index.html`);
+
+      // HACK: work around axe-core (incorrectly) requiring this to be
+      // a function (see https://github.com/dequelabs/axe-core/issues/974).
+      (config.checks as any)[0].evaluate = 'function () { return false }';
```
Comment (straker) [inline packages/puppeteer/test/axePuppeteer.test.ts:472]: I believe this should no longer be required as evaluate can be a string [in the typings](https://github.com/dequelabs/axe-core/blob/develop/axe.d.ts#L213)

```suggestion
      config.checks[0].evaluate = 'function () { return false }';
```

**Review by straker — APPROVED** [review]
Love all the deleted files. I'm fine leaving the hack in-place for this pr as it's just a typescript hack that isn't needed anymore.

File: packages/puppeteer/test/axePuppeteer.test.ts (hunk 2 above)
Comment (WilcoFiers) [reply packages/puppeteer/test/axePuppeteer.test.ts:108]: Was copied from the other source... probably shouldn't be here. I'll just take it out.

File: packages/puppeteer/test/axePuppeteer.test.ts (hunk 3 above)
Comment (WilcoFiers) [reply packages/puppeteer/test/axePuppeteer.test.ts:472]: Copied this from the previous implementation, but yeah it can be removed.

**Pull request merged**